Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -126,9 +114,27 @@ class GraphLinter: GraphLinting {
             issues.append(issue)
         }
 
+        issues.append(contentsOf: lintStaticDependencies(from: from, to: to))
+
         return issues
     }
 
+    private func lintStaticDependencies(from: TargetNode, to: TargetNode) -> [LintingIssue] {
+        guard to.target.product.isStatic, from.target.canLinkStaticProducts() else {
+            return []
+        }
+        let warning = StaticDepedencyWarning(targetNode: from,
+                                             linkingStaticTargetNode: to)
+        let (inserted, oldMember) = linkedStaticProducts.insert(warning)
```

Comment:
It was here before the PR, but I'd not hold state in the linter. If we call the linter several times with different projects that might result in invalid results.